plugins/tomcat8/src/main/java/org/radargun/service/TomcatServerService.java

```diff
@@ -0,0 +1,284 @@
+package org.radargun.service;
+
+import java.io.BufferedReader;
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.radargun.Service;
+import org.radargun.ServiceHelper;
+import org.radargun.config.Init;
+import org.radargun.config.Property;
+import org.radargun.logging.Log;
+import org.radargun.logging.LogFactory;
+import org.radargun.traits.ProvidesTrait;
+import org.radargun.utils.Base64Coder;
+
+/**
+ * A service for starting Tomcat 8 server.
+ *
+ * The username and password properties must be set so the Tomcat
+ * service can verify that the server is properly running.
+ * The server is considered properly running once the query for
+ * the following address returns "OK" in the response body:
+ *    http://bindAddress:bindHttpPort/manager/text/list
+ *
+ * The conf/tomcat-users.xml file within Tomcat must include a user
+ * with "manager-script" role and corresponding credentials.
+ *
+ * @author Martin Gencur
+ */
+@Service(doc = TomcatServerService.SERVICE_DESCRIPTION)
+public class TomcatServerService extends JavaProcessService {
+
+   protected final Log log = LogFactory.getLog(getClass());
+   protected static final String SERVICE_DESCRIPTION = "Tomcat Server";
+   private static final String JAVA_HOME = "JAVA_HOME";
```

Comment:
OK. The existing InfinispanServerService has these constants and fields related to Java too. I'll move them up to JavaProcessService as well.